WMF Framework/EventPlatformClient/SamplingController.swift

```diff
@@ -0,0 +1,133 @@
+import Foundation
+import CocoaLumberjackSwift
+
+class SamplingController: NSObject {
+
+    var appInstallId: String?
+    var sessionId: String?
+
+    /**
+     * Serial dispatch queue that enables working with properties in a thread-safe
+     * way
+     */
+    private let queue = DispatchQueue(label: "EventPlatformClientSampling-" + UUID().uuidString)
+
+    /**
+     * Cache of "in sample" / "out of sample" determination for each stream
+     *
+     * The process of determining only has to happen the first time an event is
+     * logged to a stream for which stream configuration is available. All other
+     * times `in_sample` simply returns the cached determination.
+     *
+     * Only cache determinations asynchronously via `queue.async`
+     */
+    private var samplingCache: [EPC.Stream: Bool] = [:]
+
+    /**
+     * Compute a boolean function on a random identifier
+     * - Parameter stream: name of the stream
+     * - Returns: `true` if in sample or `false` otherwise
```

Comment:
Would be good to add the new config parameter documentation here.

Reply:
Done